airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/GlueOperations.java

```diff
@@ -0,0 +1,156 @@
+package io.airbyte.integrations.destination.s3;
+
+import com.amazonaws.services.glue.AWSGlue;
+import com.amazonaws.services.glue.model.Column;
+import com.amazonaws.services.glue.model.CreateTableRequest;
+import com.amazonaws.services.glue.model.DeleteTableRequest;
+import com.amazonaws.services.glue.model.EntityNotFoundException;
+import com.amazonaws.services.glue.model.GetTableRequest;
+import com.amazonaws.services.glue.model.SerDeInfo;
+import com.amazonaws.services.glue.model.StorageDescriptor;
+import com.amazonaws.services.glue.model.TableInput;
+import com.amazonaws.services.glue.model.UpdateTableRequest;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+
+//TODO (itaseski) implement wrapper for retry logic on transient errors
+public class GlueOperations implements MetastoreOperations {
+
+    private final ObjectMapper objectMapper;
+
+    private final AWSGlue awsGlueClient;
+
+    public GlueOperations(AWSGlue awsGlueClient) {
+        Preconditions.checkArgument(awsGlueClient != null);
+        this.awsGlueClient = awsGlueClient;
+        this.objectMapper = new ObjectMapper();
+    }
+
+    //TODO (itaseski) can location change after table is created?
+    @Override
+    public void upsertTable(String databaseName, String tableName, String location, JsonNode jsonSchema) {
+        try {
+            GetTableRequest getTableRequest = new GetTableRequest()
+                .withDatabaseName(databaseName)
+                .withName(tableName);
+
+            // Will throw EntityNotFoundException if table doesn't exist
+            awsGlueClient.getTable(getTableRequest);
+
+            UpdateTableRequest updateTableRequest = new UpdateTableRequest()
+                .withDatabaseName(databaseName)
+                //TODO (itaseski) do we need to send all table inputs or the ones that are going to be changed?
```

Comment:
It should only be necessary to modify the schema of tables that are actually being modified. Existing table definitions will continue to operate as long as the underlying data is still present.

Reply:
Agreed. I'm only sending schema and location the only data that can change.